docs - AI updates (#74122)
metabase/metabase: docs/ai/overview.md

---
title: AI in Metabase
summary: Overview of all the ways you can use AI with Metabase.
---

# AI in Metabase

> There are three main ways you can use AI with your Metabase:

Reviewer: ```suggestion
AI is in Metabase optional. You can use Metabase without AI at all. But if you do want to use AI to interact with Metabase, we have you covered.

Here are the different ways to use AI with Metabase:
```
Drops "three" in case we add more. Emphasizes that all this stuff is optional.

> - [Metabot](#metabot)

Reviewer: Should we classify in-app vs not?

Should we add a link to the embeddable AI chat component?

- [MCP server](#mcp-server)
- [Agent-driven file-based development workflow](#agent-driven-development-workflow)

> Each has its own uses and

Reviewer: ```suggestion
```

## Metabot

> > See [full docs for Metabot](metabot.md).

Reviewer: Can we just have a link at the end of the section? These repeated blockquotes are a bit much.

> **Best for: daily tasks in Metabase; granular control over people's AI usage.**

Reviewer: Wonder if we should lead with description in each section, then follow with a list: Best for, Controls, etc.

Should we format these items as a list? Might be easier to parse/skim when rendered.

**Best for: daily tasks in Metabase; granular control over people's AI usage.**

> **Functionality:** Metabot is Metabase's built-in AI agent. Metabot can help you with most daily tasks around Metabase, like answering questions about your data, creating queries, generating SQL code, explaining charts, or creating Documents. See [non-exhaustive list of things Metabot can do](metabot.md#what-metabot-can-do), as well as its [limitations](metabot.md#current-limitations).

Reviewer: Can we combine best for and functionality sections? (And not call it anything, just have a short description of what it is.)

And the best for doesn't really make sense. The essential difference is in-app vs not.

Author: No it's not. For example, it's our Product Direction that if you want to create a question, you do it through Metabot, _even if you can also do it outside the app_

Author: > Can we combine best for and functionality sections?

I write for tiktok attention span

**Controls:** Metabot will only see what the person using it can see. Metabot also comes with additional permission controls and usage limits so that you control who can use which Metabot tools (e.g. chat vs SQL generation) and how many tokens they can spend.

> **Provider**: You can configure the AI provider used to process and respond people's requests:

Reviewer: ```suggestion
**Provider**:  Choose from:
```

> - Metabase's own model (available on Metabase Cloud only, comes with an additional charge)

Reviewer: ```suggestion
- Metabase's own AI Service (available as an add-on exclusively for Metabase Cloud)
```
Not a model that we trained specifically.

- Third-party models via an API key (Anthropic models only for now).

> **Plans**: available on all plans. You can only use the model hosted by Metabase on Metabase Cloud.

Reviewer: ```suggestion
**Plans**: available on all plans. You can only use Metabase's AI Service on Metabase Cloud.
```

## MCP server

> See [full docs for MCP server](mcp.md).

**Best for: askng ad-hoc, ephemeral questions; combining data from Metabase with data from other tools.**

> **Functionality**: Connect your favorite third-party AI tool - like Claude or Codex - to the Metabase MCP server. MCP server's main functionality is designed to for answering in-the-moment questions like "hey btw what's our q3 revenue?". It's also useful when combined with other MCP servers - for example, you can ask Claude a question about your customer that combines data from Metabase, Salesforce, and Zendesk.

Reviewer: ```suggestion
**Functionality**: Connect your favorite third-party AI tool to the Metabase MCP server. MCP server's main functionality is designed to for answering in-the-moment questions like "hey btw what's our q3 revenue?". The MCP server is also useful when combined with other MCP servers. For example, you can ask your AI a question about your customer that combines data from Metabase and other data sources you haven't yet added to your data warehouse.
```
I'd avoid name brands where we can, unless we specifically want to promote them.

Author: Mentioning claude and codex etc is for searcheability. The rest is to make things explicit to encourage action and usage. I read "combines data from Metabase and other data sources you haven't yet added to your data warehouse" and my eyes glaze over, i just nod along  like "sure whatever you say". I read "combines data from Metabase, Salesforce, and Zendesk" and I immediately come up with like 17 problems that this can help me solve, _even if the exact tools I use are different_

Author: But ok I'll replace it with "CRM" and "ticketing platform". I'm going to keep Claude mentions.  I do not expect people who use metabase MCP server in claude to understand what we mean by "AI tool"

Compared to the built-in Metabot, MCP server has somewhat restricted functionality (for example, it can't generate code or built transforms for now). See [the ever-expanding list of MCP server tools](mcp.md#available-tools).

> **Controls**: Metabase MCP server requires people to authenticate into Metabase, and all the responses it provides will be scoped to their permissions. However, Unlike built-in Metabot, MCP server doesn't come with granular control over which _tools_ people can use.

Reviewer: But you can enable/disable the MCP server.
